text/3579-pki.md

```diff
@@ -0,0 +1,309 @@
+
+The explanations of the threat model include  threats specifically mitigated by the quorum system and by the use of an HSM. In addition, the quorum system and the use of an HSM provide defense-in-depth protecting the root key by means more than just transparency logging, particularly since the transparency logs themselves will have a chain of trust leading up to the root key.
+
+Lastly, strictly using transparency logs with quickly rotated private key material shifts the responsibility of implementing delegation and trust to the implementing user; we wish to provide this trust inherently in our implementation for easier use.
+
+### Foregoing Delegate Keys
+
+We shall utilize delegate keys for actual day-to-day operations within the project to mitigate the threat of broad compromise. This means that in the event of, for example, crates.io being compromised and its keys needing rotation, we will not impact other areas in which the PKI is utilized. This allows us to securely isolate the impact of incidents as well as be flexible in the duration and expiration of various delegate keys within the project. Said another way: delegate keys grant us more granular control of the use cases of the PKI while protecting the PKI as a whole from compromise.
+
+### Utilizing a Multi-Signer schema for the Certificate Authority
+
+Instead of a quorum authentication model for a legacy Certificate Authority set of Root Keys, we could move to a multi-signer scheme which will be validated for all the uses cases we expect this to cover. The drawback of such an approach is that most standard scenarios in which we will want to utilize such a set of key instances do not support such a model; it would be an approach to only solve the issue of repository and crate signing. We have legitimate cases which can benefit from an SSL CA (Infra internal services, code signing, etc.) which do not support multi-signer keys. We hope this RFC will be leveraged for future uses, but consider that question out of the scope of this RFC.
+
+### Alternative Quorum Tools
+[alternative-quorum-tools]: #alternative-quorum-tools
+
+Other solutions exist for performing Quorum-based authentication for key access, which support various storage backends. The main alternative for such a standalone solution would be utilizing a quorum authentication plugin on a cloud-agnostic storage system (Such as Hashicorp Vault, or a cloud-agnostic authentication scheme atop a cloud key storage, such as AWS key store). These solutions would allow us to deploy and move independent of cloud providers; however, this comes with the added infrastructure overhead of the Infra team needing to implement and maintain such a solution. This choice was considered but thought prohibitive to implement and maintain for the current staffing available to the project. Finally, these solutions do not exist in hardware due to the need to remain cloud and hardware agnostic.
```

Comment:
For  Valut-like tools,  although it requires some extra time and effort to implementand maintain, it can not only be used as a key management tool for rust projects, but also as a secrect management tool for the entire Rust project to manage other Secrects like tokens or AK/SK in a unified manner. So I think it's possible to consider